utils/loading_cache.hh

```diff
@@ -354,7 +362,9 @@ private:
         if (!ts_ptr) {
             return;
         }
-        _lru_list.erase_and_dispose(_lru_list.iterator_to(*ts_ptr->lru_entry_ptr()), [this] (ts_value_lru_entry* p) { loading_cache::destroy_ts_value(p); });
+        ts_value_lru_entry* lru_entry_ptr = ts_ptr->lru_entry_ptr();
+        lru_list_type& entry_list = container_list(*lru_entry_ptr);
+        entry_list.erase_and_dispose(entry_list.iterator_to(*lru_entry_ptr), [this] (ts_value_lru_entry* p) { loading_cache::destroy_ts_value(p); });
```

Comment:
"not popular" == "new generation"
"popular" == "old generation".

> Does this mean that expiration (and therefore, the "reload" feature) only applies to the unpopular content

Not at all. Both reload and expiration applies (supposed to) both types of content.
What made you think otherwise?

Reply:
I thought you are doing here the expiration loop just for one of the two lists. Maybe I misread the code.